File: scratch/fix_rem.py
import os
import re

files = [
    (r'C:\dbros_app\lib\screens\write_log_page.dart', 'write'),
    (r'C:\dbros_app\lib\screens\log_list_page.dart', 'log')
]

for path, kind in files:
    with open(path, 'r', encoding='utf-8') as f:
        content = f.read()
        
    on_finish_skip = """      onFinish: () {
        Future.delayed(const Duration(milliseconds: 300), () {
          Navigator.of(context, rootNavigator: true).popUntil((route) => route.isFirst);
          mainTabEventController.add(4);
        });
        return true;
      },
      onSkip: () {
        Future.delayed(const Duration(milliseconds: 300), () {
          Navigator.of(context, rootNavigator: true).popUntil((route) => route.isFirst);
          mainTabEventController.add(4);
        });
        return true;
      },
"""
    if kind == 'write':
        # Add beforeFocus
        content = content.replace("      beforeFocus: (target) async {", on_finish_skip + "      beforeFocus: (target) async {")
        
        # Replace OCR terms
        content = content.replace('OCR(자동 인식)', '콜카드 인식(자동 입력)')
        content = content.replace('OCR(자동 인식) 이용 안내', '콜카드 인식(자동 입력) 이용 안내')
        content = content.replace('OCR 기능은', '콜카드 인식 기능은')
        content = content.replace('콜카드 이미지 자동 인식', '콜카드 인식(자동 입력)')
        
    if kind == 'log':
        # Replace ).show(context: context); for TutorialCoachMark
        content = content.replace("      opacityShadow: 0.8,\n    ).show(context: context);", "      opacityShadow: 0.8,\n" + on_finish_skip + "    ).show(context: context);")

    with open(path, 'w', encoding='utf-8') as f:
        f.write(content)
    print('Updated ' + path)

File: scratch/fix_rem.dart
import 'dart:io';

void main() {
  final files = [
    (r'C:\dbros_app\lib\screens\write_log_page.dart', 'write'),
    (r'C:\dbros_app\lib\screens\log_list_page.dart', 'log')
  ];

  for (final item in files) {
    final path = item.$1;
    final kind = item.$2;
    final file = File(path);
    if (!file.existsSync()) continue;
    
    var content = file.readAsStringSync();
        
    final onFinishSkip = '''      onFinish: () {
        Future.delayed(const Duration(milliseconds: 300), () {
          Navigator.of(context, rootNavigator: true).popUntil((route) => route.isFirst);
          mainTabEventController.add(4);
        });
        return true;
      },
      onSkip: () {
        Future.delayed(const Duration(milliseconds: 300), () {
          Navigator.of(context, rootNavigator: true).popUntil((route) => route.isFirst);
          mainTabEventController.add(4);
        });
        return true;
      },
''';

    if (kind == 'write') {
        content = content.replaceAll("      beforeFocus: (target) async {", onFinishSkip + "      beforeFocus: (target) async {");
        content = content.replaceAll('OCR(자동 인식)', '콜카드 인식(자동 입력)');
        content = content.replaceAll('OCR(자동 인식) 이용 안내', '콜카드 인식(자동 입력) 이용 안내');
        content = content.replaceAll('OCR 기능은', '콜카드 인식 기능은');
        content = content.replaceAll('콜카드 이미지 자동 인식', '콜카드 인식(자동 입력)');
    }
        
    if (kind == 'log') {
        content = content.replaceAll("      opacityShadow: 0.8,\n    ).show(context: context);", "      opacityShadow: 0.8,\n" + onFinishSkip + "    ).show(context: context);");
    }

    file.writeAsStringSync(content);
    print('Updated ' + path);
  }
}
